components/aws/sagemaker/tests/unit_tests/tests/test_hpo.py

```diff
@@ -60,4 +64,286 @@ def test_empty_string(self):
+
+    with self.assertRaises(Exception):
+      _utils.create_hyperparameter_tuning_job_request(vars(parsed_args))
+
+  def test_tags(self):
+    args = self.parser.parse_args(required_args + ['--tags', '{"key1": "val1", "key2": "val2"}'])
+    response = _utils.create_hyperparameter_tuning_job_request(vars(args))
+    self.assertIn({'Key': 'key1', 'Value': 'val1'}, response['Tags'])
+    self.assertIn({'Key': 'key2', 'Value': 'val2'}, response['Tags'])
+
+  def test_invalid_instance_type(self):
+    invalid_instance_args = required_args + ['--instance_type', 'invalid-instance']
+
+    with self.assertRaises(SystemExit):
+      self.parser.parse_args(invalid_instance_args)
+
+  def test_valid_hyperparameters(self):
+    hyperparameters_str = '{"hp1": "val1", "hp2": "val2", "hp3": "val3"}'
+
+    good_args = self.parser.parse_args(required_args + ['--static_parameters', hyperparameters_str])
+    response = _utils.create_hyperparameter_tuning_job_request(vars(good_args))
+    
+    self.assertIn('hp1', response['TrainingJobDefinition']['StaticHyperParameters'])
+    self.assertIn('hp2', response['TrainingJobDefinition']['StaticHyperParameters'])
+    self.assertIn('hp3', response['TrainingJobDefinition']['StaticHyperParameters'])
+    self.assertEqual(response['TrainingJobDefinition']['StaticHyperParameters']['hp1'], "val1")
+    self.assertEqual(response['TrainingJobDefinition']['StaticHyperParameters']['hp2'], "val2")
+    self.assertEqual(response['TrainingJobDefinition']['StaticHyperParameters']['hp3'], "val3")
+
+  def test_empty_hyperparameters(self):
+    hyperparameters_str = '{}'
+
+    good_args = self.parser.parse_args(required_args + ['--static_parameters', hyperparameters_str])
+    response = _utils.create_hyperparameter_tuning_job_request(vars(good_args))
+    
+    self.assertEqual(response['TrainingJobDefinition']['StaticHyperParameters'], {}) 
+
+  def test_object_hyperparameters(self):
+    hyperparameters_str = '{"hp1": {"innerkey": "innerval"}}'
+
+    invalid_args = self.parser.parse_args(required_args + ['--static_parameters', hyperparameters_str])
+    with self.assertRaises(Exception):
+      _utils.create_hyperparameter_tuning_job_request(vars(invalid_args))
+
+  def test_vpc_configuration(self):
+    required_vpc_args = self.parser.parse_args(required_args + ['--vpc_security_group_ids', 'sg1,sg2', '--vpc_subnets', 'subnet1,subnet2'])
+    response = _utils.create_hyperparameter_tuning_job_request(vars(required_vpc_args))
+    
+    self.assertIn('TrainingJobDefinition', response)
+    self.assertIn('VpcConfig', response['TrainingJobDefinition'])
+    self.assertIn('sg1', response['TrainingJobDefinition']['VpcConfig']['SecurityGroupIds'])
+    self.assertIn('sg2', response['TrainingJobDefinition']['VpcConfig']['SecurityGroupIds'])
+    self.assertIn('subnet1', response['TrainingJobDefinition']['VpcConfig']['Subnets'])
+    self.assertIn('subnet2', response['TrainingJobDefinition']['VpcConfig']['Subnets'])
+
```

Comment:
Can we add a test to ensure we have good support for each type of "ParameterRanges": Categorical, Continuous and Integer?